frontend/src/components/widgets/UserAvatarLink.jsx

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import Avatar from "@material-ui/core/Avatar";
+import Typography from "@material-ui/core/Typography";
+import { makeStyles } from "@material-ui/styles";
+import { getUrl as getUserBoardUrl } from "./LinkToUserBoard";
+
+const useStyles = makeStyles({
+  avatarLink: {
+    textDecoration: "none",
+  },
+  container: {
+    display: "flex",
+    alignItems: "center",
+    gap: 8,
+  },
+});
+
+export function getAvatarTextInitials(email) {
+  const customPartOfAnEmail = email.substr(0, email.indexOf("@"));
+  // For Umuzi emails - take the first letters of the firstname and lastname
+  if (email.endsWith("umuzi.org")) {
+    const [firstname, lastname] = customPartOfAnEmail.split(".");
+    return (firstname[0] + lastname[0]).toUpperCase();
+  }
+  // Other emails - take first two letters/digits
+  let firstTwoLetters = "";
+  const alphanumerals = /[a-zA-Z0-9]/;
```

Comment:
alphanuerals regex is a const and won't be changed if you were to reuse it somewhere on top of your code, you would have to redeclare, would you perhaps rather globally initialize this variable?